lib/cache.module.ts

```diff
@@ -15,7 +16,41 @@ import {
 // eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
 export abstract class Cache {}
 // eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
-export interface Cache extends CoreCache {}
+export interface Cache {
+  get: <T>(key: string) => Promise<T | null>;
+  mget: <T>(keys: string[]) => Promise<(Awaited<T> | null)[]>;
+  set: <T>(key: string, value: T, ttl?: number) => Promise<T>;
+  mset: <T>(
+    list: Array<{
+      key: string;
+      value: T;
+      ttl?: number;
+    }>,
+  ) => Promise<
+    {
+      key: string;
+      value: T;
+      ttl?: number;
+    }[]
+  >;
+  del: (key: string) => Promise<boolean>;
+  mdel: (keys: string[]) => Promise<boolean>;
+  clear: () => Promise<boolean>;
+  wrap: <T>(
+    key: string,
+    fnc: () => T | Promise<T>,
+    ttl?: number | ((value: T) => number),
+    refreshThreshold?: number,
+  ) => Promise<T>;
+  on: <E extends keyof Events>(
+    event: E,
+    listener: Events[E],
+  ) => EventEmitter<[never]>;
+  off: <E extends keyof Events>(
+    event: E,
+    listener: Events[E],
+  ) => EventEmitter<[never]>;
+}
```

Comment:
Wouldn't it be better to grab this type from `cache-manager` using `ReturnType` like below?

```suggestion
export interface Cache {
  get: <T>(key: string) => Promise<T | null>;
  mget: <T>(keys: string[]) => Promise<(Awaited<T> | null)[]>;
  set: <T>(key: string, value: T, ttl?: number) => Promise<T>;
  mset: <T>(
    list: Array<{
      key: string;
      value: T;
      ttl?: number;
    }>,
  ) => Promise<
    {
      key: string;
      value: T;
      ttl?: number;
    }[]
  >;
  del: (key: string) => Promise<boolean>;
  mdel: (keys: string[]) => Promise<boolean>;
  clear: () => Promise<boolean>;
  wrap: <T>(
    key: string,
    fnc: () => T | Promise<T>,
    ttl?: number | ((value: T) => number),
    refreshThreshold?: number,
  ) => Promise<T>;
  on: <E extends keyof Events>(
    event: E,
    listener: Events[E],
  ) => EventEmitter<[never]>;
  off: <E extends keyof Events>(
    event: E,
    listener: Events[E],
  ) => EventEmitter<[never]>;
}
```
```ts
export interface Cache extends ReturnType<createCache>
```

Reply:
Indeed, this is a better approach, and I will make revisions later. I'm sorry that I haven't studied TS before